### PR TITLE
KOGITO-8363:  Extract common language service related code into a new package

### PR DESCRIPTION
**Pull request by fantonangeli**
Jira: https://issues.redhat.com/browse/KOGITO-8363

**Description**

We have duplicated code on Serverless Workflow and Dashbuilder Language Services that can be extracted into a new package and potentially be reused by other applications.

Everything that is common should be extracted to this new package, whereas specific code must still remain on each individual package (Serverless Workflow and Dashbuilder LS).

### CODE REVIEW

File: packages/dashbuilder/package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "**/dist/*"
   ],
   "scripts": {
-    "build:dev": "run-script-os && pnpm copy:components && pnpm build:dev:runtime",
+    "build:dev": "",
```
Comment (caponetto) [inline packages/dashbuilder/package.json]: A small note to revert this change.

**Comment by tiagobento** [comment]
@fantonangeli @caponetto Can we come up with a more descriptive name for `editor-language-service`? `Editor` is a very generic term for us, and this new package seems to be related to JSON/YAML-based Editors only...

File: packages/dashbuilder/package.json (hunk 1 above)
Comment (fantonangeli) [reply packages/dashbuilder/package.json]: Thank you @caponetto , I committed this by mistake

**Comment by fantonangeli** [comment]
> @fantonangeli @caponetto Can we come up with a more descriptive name for `editor-language-service`? `Editor` is a very generic term for us, and this new package seems to be related to JSON/YAML-based Editors only...

@tiagobento I agree with you that is a very generic name.
At the moment it only supports JSON and YAML languages, but we made it extensible for others' language support in the future.
This package is not directly related to SWF and Dashbuilder and can be used even in packages outside of `kie-tools` repository, but it's always related to Monaco/VSCode. 

If you can suggest a better name I will be happy to rename the package after I finished to extract the code from SWF and Dashbuilder Language Services.

**Comment by tiagobento** [comment]
@fantonangeli Hmmm... I'm struggling to see how this package can be generic to more things other than JSON and YAML. Can you please make a simple list of high-level feature that this new "common" package would provide to other "specific-language-service" packages?

**Comment by fantonangeli** [comment]
> @fantonangeli Hmmm... I'm struggling to see how this package can be generic to more things other than JSON and YAML. Can you please make a simple list of high-level feature that this new "common" package would provide to other "specific-language-service" packages?

@tiagobento 
With @caponetto we were thinking to a rename the package to `json-yaml-language-service`.
Please let me know if you have a better name to suggest.

I wrote in the README the list of the features provided in this package: https://github.com/kiegroup/kie-tools/pull/1565/commits/7145eedbab38470b9d0a653ab8b3de1b6fe0b5cf

**Comment by tiagobento** [comment]
@fantonangeli Thanks a lot! I thin it makes sense to rename it to `json-yaml-language-service`! Good naming! :)

**Comment by fantonangeli** [comment]
Currently, I'm only missing tests for the classes to create Code Completion, Code Lenses and Diagnostic.
Those tests are already in `serverless-workflow-language-service` package, but porting them to `json-yaml-language-service` requires making a fully functional `TestLanguageService` in the test folder.
Please let me know WDYT or if this is unnecessary.

**Comment by caponetto** [comment]
Hi @tomasdavidorg, this PR is meant to reorganize the language service packages and extract common code into a new package. So it does not include any new features. Anyway, we would appreciate your help in executing sanity checks around code completion, code lenses, validation messages, etc., for the serverless workflow and Dashbuilder editors. Thanks in advance.

File: packages/serverless-workflow-language-service/src/api/index.ts [hunk 2]
```diff
@@ -14,13 +14,10 @@
...
-/**
- * File language for an opened file in the text editor.
- */
-export enum FileLanguage {
-  JSON = "json",
-  YAML = "yaml",
-}
+import {
+  FileLanguage,
+  getFileLanguage as getFileLanguageEditorLS,
```
Comment (caponetto) [inline packages/serverless-workflow-language-service/src/api/index.ts]: Can't we use `getFileLanguage` or rename it to `getFileLanguageEditorLS`?

File: packages/serverless-workflow-text-editor/src/editor/textEditor/SwfTextEditorController.ts [hunk 3]
```diff
@@ -82,8 +89,8 @@ export class SwfTextEditorController implements SwfTextEditorApi {
     });
   }
 
-  private get swfOffsetsApi(): SwfJsonOffsets | SwfYamlOffsets {
-    return this.language === FileLanguage.YAML ? this.swfYamlOffsets : this.swfJsonOffsets;
+  private get swfLanguageServiceClass() {
```
Comment (caponetto) [inline packages/serverless-workflow-text-editor/src/editor/textEditor/SwfTextEditorController.ts]: Is this getter being used?

File: packages/dashbuilder-language-service/src/channel/DashbuilderLanguageServiceCodeCompletion.ts [hunk 4]
```diff
@@ -13,57 +13,39 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { CompletionItem, CompletionItemKind, InsertTextFormat, Position, Range } from "vscode-languageserver-types";
-import { dashbuilderCompletion } from "../assets/code-completions/";
-import { DashbuilderLsNode, TranslateArgs } from "./types";
+import {
```
Comment (caponetto) [inline packages/dashbuilder-language-service/src/channel/DashbuilderLanguageServiceCodeCompletion.ts:16]: Can you please clean up unused imports on this file?

File: packages/json-yaml-language-service/jest.config.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  moduleDirectories: ["node_modules"],
+  moduleFileExtensions: ["js", "jsx", "ts", "tsx"],
+  testRegex: "/tests/.*\\.test\\.(jsx?|tsx?)$",
+  transform: {
+    "^.+\\.jsx?$": ["babel-jest", { presets: [["@babel/env", { targets: { node: "current" } }], "@babel/react"] }],
+    "^.+\\.tsx?$": "ts-jest",
+  },
+  moduleNameMapper: {
+    "\\.(css|less|sass|scss)$": "<rootDir>/tests/__mocks__/styleMock.js",
+  },
```
Comment (caponetto) [inline packages/json-yaml-language-service/jest.config.js]: I believe this is unnecessary for this package.

**Review by caponetto — COMMENTED** [review]
Hey @fantonangeli, well done! Thank you for this effort!
I've left some very minor comments inline.
I'm still validating it but overall looks very good!

**Comment by fantonangeli** [comment]
Hey @caponetto , thank you very much for your review and for your comments.

**Comment by tomasdavidorg** [comment]
@caponetto Starting the testing now. Will let you know if I find any issue.

**Review by tomasdavidorg — APPROVED** [review]
Looks good to me. I tested json and yaml editors as Vscode extension and the Web tools and I didn't find any issue. 
@barboras7 Could you please take a look on the dashbuilder editor?

**Review by caponetto — APPROVED** [review]

**Comment by caponetto** [comment]
We should probably wait for https://github.com/kiegroup/kie-tools/pull/1588 be merged first.

**Review by caponetto — APPROVED** [review]

**Comment by barboras7** [comment]
@fantonangeli sorry for the delay, public holiday + sick day :face_with_spiral_eyes: Anyway, I found no issues, thanks for the PR!

**Review by barboras7 — APPROVED** [review]
